gradle-sls-packaging/src/test/groovy/com/palantir/gradle/dist/tasks/CreateManifestTaskIntegrationSpec.groovy

```diff
@@ -285,6 +310,52 @@ class CreateManifestTaskIntegrationSpec extends GradleIntegrationSpec {
         manifest.get("extensions").get("product-dependencies").isEmpty()
     }
 
+    def 'Mark as optional product dependencies'() {
+        setup:
+        buildFile << """
+            dependencies {
+                runtime 'a:a:1.0'
+            }
+
+            tasks.createManifest {
+                optionalProductIds = [
+                    new com.palantir.gradle.dist.ProductId("group:name"), 
+                    new com.palantir.gradle.dist.ProductId("group:name2")
```

Comment:
Seems pretty grim that you need to do `new com.palantir.gradle.dist.ProductId(` - perhaps it should just be `String` instead so it's a bit easier to use? I realise the `ignoredProductIds` is like that, but a search on our internal code shows no one using `ignoredProductIds` except to set it to `[]`, so I'm not too worried about diverging with it.

Reply:
(take a string but still use ProductId internally, that is)

Reply:
I am not sure I get the wanted change. This complicates the logic of either the Extension or the CreateManifestTask for the conversion and storing of the properties, while this is just a test. In practice it's the `optionalProductDependency` that is used.

Reply:
Ah, missed the difference between `optionalProductIds` and `optionalProductDependency` - in that case please can we use `optionalProductDependency` in the test, just so that we're testing the API we expect people to use?

Reply:
It does not work if I use it in the test, I tried. I think it's because it is looking for it in the "createManifest" task while those functions are created in the extension

Reply:
Ah, I see this is testing the generic `createManifest` task rather than the service/asset/recommended-pdeps plugins.